### PR TITLE
feat: add is_init() to glean_usage queries

### PR DESCRIPTION
**Pull request by kik-kik**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-3645)

### CODE REVIEW

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+
+{% raw %}
+{% if is_init() %}
+{% endraw %}
+SELECT
+    *
+  FROM
+    `{{ project_id }}.{{ app_name }}.metrics_clients_daily` AS m
+  WHERE
+    FALSE
```
Comment (scholtzan) [inline sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:11]: We need to get the right metrics fields here:

```suggestion
SELECT
  DATE(@submission_date) AS submission_date,
  _current.client_id,
  _current.sample_id,
  _current.normalized_channel,
  _current.n_metrics_ping,
  _current.days_sent_metrics_ping_bits AS days_sent_metrics_ping_bits,
  {% if app_name in metrics -%}
  {% for metric in metrics[app_name] -%}
    _current.{{metric}} AS {{metric}},
  {% endfor -%}
  {% endif -%}
FROM
  `{{ project_id }}.{{ app_name }}.metrics_clients_daily` AS _current
WHERE FALSE
```

File: sql_generators/glean_usage/templates/baseline_clients_last_seen.view.sql
Comment (sean-rose) [inline sql_generators/glean_usage/templates/baseline_clients_last_seen.view.sql:1]: The changes to this view and some of the changes to `clients_last_seen_joined.query.sql` seem like they're unrelated to the purpose of adding `is_init()`, and are kind of bleeding over from #5434.  I'd suggest removing these unrelated changes.

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql [hunk 2]
```diff
@@ -1,3 +1,18 @@
+{{ header }}
+
+{% raw %}
+{% if is_init() %}
+{% endraw %}
```
Comment (sean-rose) [inline sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:5]: For consistency (e.g. you specifically changed `baseline_clients_last_seen_v1.query.sql` to switch to this single-line format).
```suggestion
{% raw %}{% if is_init() %}{% endraw %}
```
(ditto for the `{% else %}` and `{% endif %}` parts)

**Comment by kik-kik** [comment]
Sorry for the delay @sean-rose @scholtzan , I made the changes you suggested in the PR. I wonder though if merging this makes sense of if I should just drop these changes? WDYT?

**Review by scholtzan — COMMENTED** [review]
r+wc 
Yes, this is still something we'd want. Sometimes stage deploys fail since some of these tables don't have explicit schemas and can't be initialized due to the self references

File: sql_generators/glean_usage/templates/baseline_clients_last_seen_v1.query.sql [hunk 3]
```diff
@@ -1,9 +1,5 @@
 {{ header }}
-
-{% raw %}
-{% if is_init() %}
-{% endraw %}
-
+{% raw %}{% if is_init() %}{% endraw %}
 SELECT
   CAST(NULL AS INT64) AS days_seen_bits,
```
Comment (sean-rose) [inline sql_generators/glean_usage/templates/baseline_clients_last_seen_v1.query.sql:4]: For this init query to produce the intended column order it should select `submission_date` as the first column and instead of `*` use `* EXCEPT (submission_date)`.

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql [hunk 4]
```diff
@@ -1,3 +1,25 @@
+{{ header }}
+
+{% raw %}{% if is_init() %}{% endraw %}
+SELECT
+  DATE(@submission_date) AS submission_date,
```
Comment (sean-rose) [inline sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:5]: I don't think using the `@submission_date` parameter in the init use case makes sense.
```suggestion
  submission_date,
```

File: sql_generators/glean_usage/templates/clients_last_seen_joined.query.sql [hunk 5]
```diff
@@ -1,20 +1,27 @@
+{{ header }}
+
 WITH baseline AS (
   SELECT
     *
   FROM
     `{{ project_id }}.{{ app_name }}.baseline_clients_last_seen`
   WHERE
     submission_date = @submission_date
+    {% raw %}{% endif %}{% endraw %}
```
[the remark this answers is not in the document]
Comment (sean-rose) [reply sql_generators/glean_usage/templates/clients_last_seen_joined.query.sql:10]: I'm guessing what's intended here is a full `if`/`else` statement which results in `WHERE FALSE` for the init use case.

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql [hunk 6]
```diff
@@ -1,3 +1,25 @@
...
+  {% for metric in metrics[app_name] -%}
+    _current.{{metric}} AS {{metric}},
+  {% endfor -%}
+  {% endif -%}
+FROM
+  `{{ project_id }}.{{ app_name }}.metrics_clients_daily` AS _current
+WHERE FALSE
+{% raw %}
+{% else %}
+{% endraw %}
```
Comment (sean-rose) [inline sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:21]:
```suggestion
{% raw %}{% else %}{% endraw %}
```

File: sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql [hunk 7]
```diff
@@ -46,3 +68,7 @@ FULL JOIN
     OR (_previous.normalized_channel IS NULL AND _current.normalized_channel IS NULL)
   )
 WHERE _current.client_id IS NOT NULL
+
+{% raw %}
+{% endif %}
+{% endraw %}
```
Comment (sean-rose) [inline sql_generators/glean_usage/templates/metrics_clients_last_seen.query.sql:74]:
```suggestion
{% raw %}{% endif %}{% endraw %}
```

**Comment by kik-kik** [comment]
stale, closing.
